src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -178,6 +183,9 @@
     private List<PluginGerritEvent> triggerOnEvents;
     private boolean dynamicTriggerConfiguration;
     private String triggerConfigURL;
+    private final EnvVars envVars;
```

Comment:
This is no good, when the project is saved these fields will be serialized with the trigger and take up unnecessary disk space.

I think a `ThreadLocal` `LoadingCache` or something is better.